http.go

```diff
@@ -353,6 +353,8 @@ func RetryRequest(ctx context.Context, r *http.Request, opts RetryRequestOptions
 
 		if resp.StatusCode >= 200 && resp.StatusCode < 300 {
 			if opts.RetryOnInvalidBody {
+				ogBody := resp.Body
+				defer ogBody.Close()
```

Comment:
nit: I like the name, but I don't know if current slang will carry over to the future, can we call this `originalBody`?